### PR TITLE
Added test for leader proxying

### PR DESCRIPTION
**Pull request by dperny**
**\- What I did**
Added test to check leader proxying

**\- How I did it**
Create 3 managers. Start a service by making a request to each manager. Check that all services are running, which means requests were correctly proxied to leader.

**\- How to verify it**
`TESTFLAGS=-check.f DockerSwarmSuite.TestApiSwarmLeaderProxy make test-integration-cli`

**\- Description for the changelog**

Added test to check leader proxying

**\- A picture of a cute animal (not mandatory but encouraged)**
I was going to try to get a picture of a pokemon but there are none around right now.

Signed-off-by: Drew Erny drew.erny@docker.com

### CODE REVIEW

**Comment by dperny** [comment]
Build failing on gccgo, but i do not know why. Flaky build, I think.

File: integration-cli/docker_api_swarm_test.go [hunk 1]
```diff
@@ -446,6 +446,28 @@ func (s *DockerSwarmSuite) TestApiSwarmServicesStateReporting(c *check.C) {
...
+	d2.createService(c, simpleTestService, func(s *swarm.Service) {
+		s.Spec.Name = "test2"
+	})
+	d3.createService(c, simpleTestService, func(s *swarm.Service) {
+		s.Spec.Name = "test3"
+	})
+
+	// 3 services should be started now, because the requests were proxied to leader
+	// simpleTestService only has 1 replica, so there should be 3 containers
+	waitAndAssert(c, defaultReconciliationTimeout, reducedCheck(sumAsIntegers, d1.checkActiveContainerCount, d2.checkActiveContainerCount, d3.checkActiveContainerCount), checker.Equals, 3)
```
Comment (tonistiigi) [inline integration-cli/docker_api_swarm_test.go]: I think better test would be to check that all that all daemons see all these 3 services. Actually starting containers on service creation is already covered.

Comment (dperny) [inline integration-cli/docker_api_swarm_test.go]: So, list services from all 3 managers, and make sure all 3 services are in each list?

Comment (abronan) [inline integration-cli/docker_api_swarm_test.go]: @dperny Yes :)

**Comment by dperny** [comment]
Pushed new test with suggested change @tonistiigi @abronan

**Comment by abronan** [comment]
LGTM

**Comment by tonistiigi** [comment]
LGTM

**Comment by icecrime** [comment]
LGTM

**Pull request merged**